### PR TITLE
Fix not able to open workspace with nightly code

### PR DESCRIPTION
**Pull request by mustard-mh**
## Description

Changes on openvscode-server side https://github.com/gitpod-io/openvscode-server/commit/2915f0a1347a12df8b19641d99235395705efb53

TODO:
- [x] Revert debug commit https://github.com/gitpod-io/gitpod/pull/20029/commits/be496bc04a55152b0e29cbc3ea36d3ab0b374195

## Related Issue(s)

Fixes ENT-453, ENT-455

## How to test

**Night code is working again**
- With debug commit https://github.com/gitpod-io/gitpod/pull/20029/commits/be496bc04a55152b0e29cbc3ea36d3ab0b374195
- You should be able to open workspace with **latest Editor** in preview env

**Switch editor version is working**
- Open a workspace with latest editor
- Close workspace
- Update setting to use `stable` editor
- Restart workspace
- Workspace should use stable editor

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - hw-code-latest</li>
	<li><b>🔗 URL</b> - <a href="https://hw-code-latest.preview.gitpod-dev.com/workspaces" target="_blank">hw-code-latest.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - hw-code-latest-gha.27420</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-hw-code-latest%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-dev-preview" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [x] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

**Review by filiptronicek — APPROVED** [review]
Happy you got it to work again, nice job! Let's :shipit:

**Comment by mustard-mh** [comment]
Debug commit reverted

/unhold

File: components/server/src/workspace/workspace-starter.ts [hunk 1]
```diff
@@ -307,7 +307,10 @@ export class WorkspaceStarter {
                     ideSettings = {
                         ...ideSettings,
                         defaultIde: ideConfig.ide,
-                        useLatestVersion: !!ideConfig.useLatest,
+                        useLatestVersion:
+                            ideSettings?.useLatestVersion ??
+                            user.additionalData?.ideSettings?.useLatestVersion ??
+                            !!ideConfig.useLatest,
```
Comment (filiptronicek) [inline components/server/src/workspace/workspace-starter.ts:313]: BTW, are we fixing something here? It looks like we're changing the prioritization of what gets to dictate whether the latest version of the editor is used, so I'm curious why it's needed 👀

Comment (mustard-mh) [reply components/server/src/workspace/workspace-starter.ts:313]: See https://linear.app/gitpod/issue/ENT-455

Comment (filiptronicek) [reply components/server/src/workspace/workspace-starter.ts:313]: Oh nice, thanks!

**Pull request merged**